src/compiler/transformers/declarations.ts

```diff
@@ -782,12 +782,23 @@ namespace ts {
                     decl.modifiers,
                     /*importClause*/ undefined,
                     rewriteModuleSpecifier(decl, decl.moduleSpecifier),
-                    getResolutionModeOverrideForClause(decl.assertClause) ? decl.assertClause : undefined
+                    getResolutionModeOverrideForClauseInNightly(decl.assertClause)
                 );
             }
             // Nothing visible
         }
 
+        function getResolutionModeOverrideForClauseInNightly(assertClause: AssertClause | undefined) {

```

Comment:
Should this be getting called in the `checker`? Seems like you'll only get an error in `.d.ts` generation scenarios